src/AutoRest.CSharp/LowLevel/Generation/LowLevelModelWriter.cs

```diff
@@ -48,9 +49,19 @@ private static void WritePublicConstructor(CodeWriter writer, ConstructorSignatu
             {
                 var initializedFields = WriteFieldsInitialization(writer, signature, model);
                 // TODO: Add IReadOnlyDictionary
-                foreach (var field in model.Fields.Where(f => !initializedFields.Contains(f) && TypeFactory.IsReadOnlyList(f.Type)))
+                foreach (var field in model.Fields.Where(f => !initializedFields.Contains(f)))
                 {
-                    writer.Line($"{field.Name:I} = new List<{field.Type.Arguments[0]}>(0).AsReadOnly();");
+                    if (TypeFactory.IsList(field.Type))
+                    {
+                        if (TypeFactory.IsReadOnlyList(field.Type))
+                        {
+                            writer.Line($"{field.Name:I} = new List<{field.Type.Arguments[0]}>(0).AsReadOnly();");
+                        }
+                        else if (!field.IsRequired)
+                        {
+                            writer.Line($"{field.Name:I} = {Constant.NewInstanceOf(TypeFactory.GetPropertyImplementationType(field.Type)).GetConstantFormattable()};");
```

Comment:
I think list and dictionary properties should always have default values. That way we simplify logic for constructor writers.